sourcegraph/sourcegraph
# Code Insights: Add new insight error UI V2

Closes https://github.com/sourcegraph/sourcegraph/issues/44961
[Designs ](https://www.figma.com/file/aMO6sPoUHDUUcqa25wFnWl/Surface-timeout-limit-hit-to-end-users-in-the-UI-%2339969-Ready-for-dev?node-id=2034%3A5789&t=maBwfj5IA1hx4O3n-0)

## Background

This PR adds new insight error UI. Previously we had only timeout errors UI and it was pretty limited we didn't show any timeout error UI information (such as list of points that are failed with timeout errors). In this PR we list these points and also we add handling for all other error types (such as Generic error points). See this comment for more details https://github.com/sourcegraph/sourcegraph/issues/44961#issuecomment-1339926367

| Before | After |
| ------- | ------ | 
| <img width="432" alt="Screenshot 2022-12-07 at 02 42 37" src="https://user-images.githubusercontent.com/18492575/206098137-ec48e9d8-c3a1-43d0-9879-a2bf429c46a9.png"> | <img width="479" alt="Screenshot 2022-12-07 at 02 42 13" src="https://user-images.githubusercontent.com/18492575/206098194-b4f1f261-37b2-4f89-a4cc-9f72a18dfeee.png"> |

## Design review 
- Since I don't see timeout or generic errors on the k8s staging I suggest to use storybook story to check new error UI. For example, in [this storybook demo](https://www.chromatic.com/component?appId=5f0f381c0e50750022dc6bf7&csfId=web-insights-smartinsightsviewgridexample&buildNumber=56072&k=63902e94886b4102a13bb1d3-576-interactive-true&h=4&b=-1)
- @AlicjaSuska Grey error UI icons seemed very bold, In this PR I used icon with --warning (standard color from our palette for warning UI) let me know if we still want to stick with grey one. 
- Ideally I would add hover effect (if you hover point in the error popover we hover this point on the chart) but since we don't have time and this PR already has a lot of changes I suggest iterate on this later. 

## For reviewers 
Please don't look at PR diff size, most of this size is mock changes (I had to change our mocks since I changed the data insight query), Without mocks this PR is standard ~400-450 lines of code. 

## Test plan
- Go to the "Smart Insights View Grid Example" story 
- Check that layout UI next to the legend items looks correct
- Check that layout UI next to the insight title also looks correct 



## App preview:

- [Web](https://sg-web-vk-insight-error-ui.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
